packages/web3-react/src/utils/connector.ts

```diff
@@ -139,17 +155,32 @@ const injectedDisconnect = async (signerProvider: SignerProvider): Promise<void>
   return await (signerProvider as AlephiumWindowObject).disconnect()
 }
 
-const injectedAutoConnect = async (options: ConnectOptions): Promise<Account | undefined> => {
+const injectedAutoConnect = async (options: InjectedAutoConnectOptions): Promise<Account | undefined> => {
   try {
-    const windowAlephium = await getDefaultAlephiumWallet()
-    const enabledAccount = await windowAlephium?.enableIfConnected({ ...options, networkId: options.network })
-
-    if (windowAlephium && enabledAccount) {
-      await options.onConnected({ account: enabledAccount, signerProvider: windowAlephium })
-      setLastConnectedAccount('injected', enabledAccount, options.network)
+    const allProviders = options.allInjectedProviders ?? []
+    if (allProviders.length === 0) {
+      const windowAlephium = await getDefaultAlephiumWallet()
+      if (windowAlephium !== undefined) {
+        allProviders.push(windowAlephium)
+      }
     }
-
-    return enabledAccount
+    const enableOptions = {
+      addressGroup: options.addressGroup,
+      keyType: options.keyType,
+      networkId: options.network,
+      onDisconnected: undefined as any
+    }
+    for (const provider of allProviders) {
+      const enabledAccount = await provider.enableIfConnected(enableOptions as any)
+      if (enabledAccount) {
+        await options.onConnected({ account: enabledAccount, signerProvider: provider })
+        setLastConnectedAccount('injected', enabledAccount, options.network)
+        // eslint-disable-next-line
+        ;(provider as any)['onDisconnected'] = options.onDisconnected
```

Comment:
This is to work around the `enableIfConnected` serialization issue in onekey wallet: https://github.com/OneKeyHQ/cross-inpage-provider/blob/23a0e8e3ef2ace42e8a0359be49ddfd2a653d281/packages/providers/onekey-alph-provider/src/OnekeyAlphProvider.ts#L147, it should handle the callback in the same way as `unsafeEnable`.